Services/Mentor/OJS.Services.Mentor.Business/Implementations/MentorBusinessService.cs

```diff
@@ -0,0 +1,480 @@
+        if (model.ConversationMessages.All(cm => cm.Role != ChatMessageRole.System))
+        {
+            /*
+             *  In the first version of the mentor, there will be only a single
+             *  template in the database. This is why we can be sure that
+             *  .FirstOrDefaultAsync() will always return a template. This
+             *  will be changed in the future.
+             */
+            var template = await this.mentorPromptTemplateData
+                .GetQuery()
+                .FirstOrDefaultAsync();
+
+            var wordFiles = model.ProblemResources
+                .Where(pr =>
+                       pr is { File: not null, FileExtension: Docx } ||
+                       pr.Link is not null && pr.Link.Split('.').Last().Equals(Docx, StringComparison.Ordinal))
+                .ToList();
+
+            /*
+             *  There are 2 cases when it comes to document retrieval:
+             *  1. The problem has its own problem resource ( Word file ) ( e.g. online / onsite exam ).
+             *  2. All the problems' descriptions are in a single Word file ( e.g. Lab / Exercise ).
+             */
+            var problemsDescription = wordFiles.FirstOrDefault(pr => pr.ProblemId == model.ProblemId) ?? wordFiles.FirstOrDefault();
+
+            if (problemsDescription is null)
+            {
+                throw new BusinessServiceException(DocumentNotFoundOrEmpty);
+            }
+
+            var file = problemsDescription.File ??
+                (problemsDescription.Link is not null
+                ? await this.DownloadDocument(problemsDescription.Link)
+                : []);
+
+            var number = GetProblemNumber(model.ProblemName);
+            var text = ExtractSectionFromDocument(file, model.ProblemName, number);
+
+            model.ConversationMessages.Add(new ConversationMessageModel
+            {
+                Content = string.Format(CultureInfo.InvariantCulture, template!.Template, model.ProblemName, text),
+                Role = ChatMessageRole.System,
+                SequenceNumber = int.MinValue,
+            });
+        }
+
+        var messagesToSend = new List<ChatMessage>();
+
+        var systemMessage = model.ConversationMessages.First(cm => cm.Role == ChatMessageRole.System);
+        systemMessage.Content = RemoveRedundantWhitespace(systemMessage.Content);
+        messagesToSend.Add(CreateChatMessage(systemMessage.Role, systemMessage.Content));
+
+        var recentMessages = model.ConversationMessages
+            .Where(cm => cm.Role != ChatMessageRole.System)
+            .OrderByDescending(cm => cm.SequenceNumber)
+            .Take(GetNumericValue(settings, nameof(MentorMessagesSentCount)))
+            .ToList();
+
+        foreach (var message in recentMessages)
+        {
+            message.Content = RemoveRedundantWhitespace(message.Content);
+        }
+
+        var mentorModel = Enum.Parse<OpenAIModels>(settings[MentorModel]).ToModelString();
```

Comment:
`Enum.Parse` will fail if the setting is not correct. Better use `Enum.TryParse` and throw the exception if not parsed, even before calling `.ToModelString()`.
Also `.ToModelString()` is better to throw `ArgumentOutOfRangeException` if the provided `Enum` is not handled correctly (as we don't expect such case with proper validation) instead of returning `null`.

Reply:
Done.